lib/js_routes.rb

```diff
@@ -216,13 +216,19 @@ def build_js(route, parent_route)
   end
 
   def route_js_arguments(route, parent_spec)
+    parts = route.parts
     required_parts = route.required_parts.clone
-    optional_parts = route.parts - required_parts
-    default_parts = route.defaults.select do |part, _|
+    optional_parts = parts - required_parts
+    default_options = route.defaults.select do |part, _|
       FILTERED_DEFAULT_PARTS.exclude?(part) && URL_OPTIONS.include?(part) || required_parts.include?(part)
     end
     [
-      required_parts, optional_parts, serialize(route.path.spec, parent_spec), default_parts
+      parts,
+      required_parts,
+      optional_parts,
+      default_options,
+      Rails.version < '4.0',
+      serialize(route.path.spec, parent_spec)
```

Comment:
Js-Routes is optimized to generate as small amount of JS as possible per rails route. That is why number of route arguments needs to be minified:

1. Pass `Rails.version < '4.0'` through configuration but a constant argument of each route.
2. You don't need to pass `parts  required_parts and optional_parts` at the same time. Leave it as it was with only required and optional parts and concat them in JS to receive `parts`.